packages/dom-expressions/src/client.js

```diff
@@ -570,7 +569,14 @@ function appendNodes(parent, array, marker = null) {
   for (let i = 0, len = array.length; i < len; i++) parent.insertBefore(array[i], marker);
 }
 
+let moved;
+
 function cleanChildren(parent, current, marker, replacement) {
+  if (!moved) {
+    moved = new WeakSet();
+    queueMicrotask(() => (moved = null));
+  }
```

Comment:
`WeakSet` is already used by hydration so doesn't hurt anything but there might be better approaches than this.

Reply:
I suggest considering a strong referenced array to be cleared in the microtask as `moved.length = 0`. As the cost of instantiating could be removed, and also a set will try to deduplicate the nodes

Reply:
I was thinking if weakset was faster than array to check for element existence. Any form of book-keeping works really. We could also keep it in the nodes but I don't know the performance side of this. Essentially we need to know if a node has changed position in the tree.